src/lib/middleware/bm_service_request.h

```diff
@@ -5,6 +5,7 @@
 
 /*!
  * @brief Callback function for bm_service_request
+ * @note Should not call bm_service_request from within this callback.
```

Comment:
Calling a request nested in the callback can cause a deadlock.  

We can add support for this when we need it